client/my-sites/email/form/mailboxes/components/new-mailbox-list/index.tsx

```diff
@@ -0,0 +1,185 @@
+		recordTracksEvent( eventName, { mailbox_count: newMailboxes.length } );
+	};
+
+	const removeMailbox = useCallback(
+		( uuid: string ) => () => {
+			const newMailboxes = mailboxes.filter(
+				( mailbox ) => mailbox.formFields.uuid.value !== uuid
+			);
+			setMailboxes( newMailboxes );
+			const eventName = isTitan
+				? 'calypso_email_titan_add_mailboxes_remove_mailbox_button_click'
+				: 'calypso_email_google_workspace_add_mailboxes_remove_mailbox_button_click';
+			recordTracksEvent( eventName, { mailbox_count: newMailboxes.length } );
+		},
+		[ isTitan, mailboxes ]
+	);
+
+	const handleCancel = () => onCancel();
+
+	const persistMailboxesToState = () => {
+		setMailboxes( [ ...mailboxes ] );
+	};
+
+	const handleSubmit = ( event: FormEvent< HTMLFormElement > ) => {
+		event.preventDefault();
+
+		mailboxes.forEach( ( mailbox ) => mailbox.validate( true ) );
+		persistMailboxesToState();
+
+		onSubmit( new MailboxOperations( mailboxes, persistMailboxesToState ) );
+	};
+
+	return (
+		<form onSubmit={ handleSubmit }>
+			<fieldset className="new-mailbox-list__main" disabled={ areButtonsBusy }>
+				{ mailboxes.map( ( mailbox, index ) => {
+					const uuid = mailbox.formFields.uuid.value;
+
+					const onFieldValueChanged = ( field: MailboxFormFieldBase< string > ) => {
+						if ( ! [ FIELD_FIRSTNAME, FIELD_NAME ].includes( field.fieldName ) ) {
+							return;
+						}
+						if ( mailbox.getIsFieldTouched( FIELD_MAILBOX ) ) {
+							return;
+						}
+
+						mailbox.setFieldValue( FIELD_MAILBOX, sanitizeMailboxValue( field.value ) );
+						mailbox.validateField( FIELD_MAILBOX );
+
+						mailbox.formFields.mailbox.dispatchState();
+					};
+
+					return (
+						<Fragment key={ 'form-' + uuid }>
+							{ index > 0 && (
+								<CardHeading
+									className="new-mailbox-list__numbered-heading"
+									tagName="h3"
+									size={ 20 }
+								>
+									{ translate( 'Mailbox %(position)s', {
+										args: { position: index + 1 },
+										comment:
+											'%(position)s is the position of the mailbox in a list, e.g. Mailbox 1, Mailbox 2, etc',
+									} ) }
+								</CardHeading>
+							) }
+							<MailboxFormWrapper mailbox={ mailbox } onFieldValueChanged={ onFieldValueChanged }>
+								<>
+									<div className="new-mailbox-list__actions">
+										{ index > 0 && (
+											<Button onClick={ removeMailbox( uuid ) } busy={ areButtonsBusy }>
+												<Gridicon icon="trash" />
+												<span>{ translate( 'Remove this mailbox' ) }</span>
+											</Button>
+										) }
+									</div>
+
+									<hr className="new-mailbox-list__separator" />
+								</>
+							</MailboxFormWrapper>
+						</Fragment>
+					);
+				} ) }
+
+				<div
+					className={ classNames( 'new-mailbox-list__supplied-actions', {
+						'not-show-add-new-mailbox': ! showAddNewMailboxButton,
+					} ) }
+				>
+					{ showAddNewMailboxButton && (
+						<Button onClick={ addMailbox } busy={ areButtonsBusy }>
```

Comment:
Same thing regarding `disabled` vs `busy` as for the "Remove this mailbox" button.